UnitsNet/CustomCode/UnitValueAbbreviationLookup.cs

```diff
@@ -63,10 +63,9 @@ internal List<string> GetAbbreviationsForUnit(int unit)
 
         internal List<int> GetUnitsForAbbreviation(string abbreviation)
         {
-            if(!abbreviationToUnitMap.TryGetValue(abbreviation, out var units))
-                abbreviationToUnitMap[abbreviation] = units = new List<int>();
-
-            return units.Distinct().ToList();
+            return abbreviationToUnitMap
+                .Where(x => x.Key.Equals(abbreviation, StringComparison.OrdinalIgnoreCase))
+                .SelectMany(x => x.Value).Distinct().ToList();
```

Comment:
Put Distinct and ToList on each a new line.
The exception is if the total chain is so short the entire thing can stay on a single line and still be readable.

Reply:
This will be slower than since it's doing a linear search rather than a dictionary lookup. Perhaps we could achieve both speed and case insensitivity by just doing a ToLower when storing or looking up things instead?

Reply:
Ah, good point, I missed that. Yup, I agree with you.

Reply:
If I change the abbreviationToUnitMap to store the lower case version, then it breaks several unit tests. Specifically where we are using this to get the abbreviation on ToString.
10 °C => 10 °c (on .ToString(CultureInfo.InvariantCulture, "{0:0} {1}"))
We could add another abbreviationToUnitMap, or possibly change one of the types of the dictionary to accommodate having a dictionary lookup. Do you think it's worth the added complexity/memory space?